### PR TITLE
Revamping the way the channel shutdown happens.

### PR DESCRIPTION
**Pull request by sduskis**
Adding additional components to the shutdown process, such as the NioEventLoopGroup.
Removing ClientCloseHandler and CloseableChannel in favor of the standard Closeable interface.
Simplified the use of ReconnectingChannel and ChannelPool in BigtableChannels

### CODE REVIEW

**Comment by sduskis** [comment]
#359 was closed while doing a rebase.  This is the same code.

File: bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableChannels.java [hunk 1]
```diff
@@ -66,9 +66,9 @@
   private static final String POST_RETRY_REPORT_ENTRY = "PostRetry";
   /** Number of threads to use to initiate retry calls */
   private static final int RETRY_THREAD_COUNT = 5;
-  
+
   /** Number of seconds to wait for a termination before trying again. */
-  private static final int CHANNEL_TERMINATE_WAIT_SECONDS = 5;
+  public static final int CHANNEL_TERMINATE_WAIT_MS = 5;
```
Comment (mgarolera) [inline bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableChannels.java]: 5 MS or 5000 MS?

File: bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/ReconnectingChannel.java [hunk 2]
```diff
@@ -29,25 +31,27 @@
...
 
 /**
  * A ClosableChannel that refreshes itself based on a user supplied timeout.
  */
-public class ReconnectingChannel extends CloseableChannel {
+public class ReconnectingChannel extends Channel implements Closeable {
 
   protected static final Logger log = Logger.getLogger(ChannelPool.class.getName());
-  protected static final long CLOSE_WAIT_TIME = 5000;
+  protected static final long CLOSE_WAIT_TIME = BigtableChannels.CHANNEL_TERMINATE_WAIT_MS;
```
Comment (mgarolera) [inline bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/ReconnectingChannel.java]: Could you add the unit to CLOSE_WAIT_TIME ?

File: bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableGrpcClient.java [hunk 3]
```diff
@@ -159,12 +162,12 @@ public static BigtableClient createClient(
...
   public static final int SCANNER_BUFFER_SIZE = 32;
   
-  private final CloseableChannel bigtableChannel;
+  private final Channel bigtableChannel;
   private final ExecutorService executorService;
   private final BigtableGrpcClientOptions clientOptions;
 
   public BigtableGrpcClient(
-      CloseableChannel closeableChannel,
+      Channel closeableChannel,
```
Comment (mgarolera) [inline bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableGrpcClient.java]: closeableChannel -> channel

File: bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableGrpcClient.java [hunk 4]
```diff
@@ -135,7 +134,12 @@ public void onCompleted() {
   }
 
   /**
-   * Factory method to create a Cloud Bigtable data client.
+   * Create a new BigtableGrpcClient. Note, the caller needs to take responsibility of
```
Comment (mgarolera) [inline bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableGrpcClient.java]: I think this comment should be in the constructor of BigtableGrpcClient, at least this is where it is for BigtableTableAdminGrpcClient. Also, it doesnt appear for BigtableClusterAdminGrpcClient

**Comment by mgarolera** [comment]
LGTM

**Pull request merged**